CRM/Core/CodeGen/OptionGroup.php

```diff
@@ -0,0 +1,137 @@
+<?php
+class CRM_Core_CodeGen_OptionGroup {
+
+  /**
+   * @var array
+   * @internal
+   */
+  public $metadata = [
+    'is_active' => 1,
+    'is_reserved' => 1,
+    'option_value_fields' => 'name,label,description',
+  ];
+
+  protected $defaults = [
+    'color' => NULL,
+    'component_id' => NULL,
+    'description' => NULL,
+    'domain_id' => NULL,
+    'filter' => 0,
+    'grouping' => NULL,
+    'icon' => NULL,
+    'is_active' => 1,
+    'is_default' => 0,
+    'is_optgroup' => 0,
+    'is_reserved' => 0,
+    'visibility_id' => NULL,
+  ];
+
+  protected $var;
+
+  protected $rows = [];
+
+  /**
+   * @var string
+   * @internal
+   */
+  public $historicalId;
```

Comment:
Arg - more freud-case. For years I thought we were removing freud case but it's going back in

Reply:
It enters through the subconscious 🧠

Reply:
Yeah - I have been resolving all those conflicts on `participantId` vs `participantID` on the non-freudian variant over time - but regression is a freudian concept I believe.....

Reply:
I don't know what freud-case is... Was it discussed in "Civilization and its Discontents"? 

But FWIW, I don't like the existence or name of the property. A better description might be "sort key". (I had to add it to make sure the OG's ended up with the same order/naming.)

Reply:
Freud case is when you camel case the word 'id'  & get `Id` - so  when reading code  'historical id' is different from  'historical ID'

Reply:
@eileenmcnaughton so are you saying our variables should be named like `$superEgoID` instead of `$superEgoId`?

Reply:
@colemanw it depends on whether you are referring to an id or an ID - for example my boys have Student IDs and Student ids - the former is useful - the latter comes out when they are hangry